presidio-anonymizer/presidio_anonymizer/entities/engine/pii_entity.py

```diff
@@ -0,0 +1,43 @@
+import logging
+from abc import ABC
+
+from presidio_anonymizer.entities import InvalidParamException
+from presidio_anonymizer.services.validators import validate_parameter_not_empty, \
+    validate_parameter_exists, validate_type
+
+
+class PIIEntity(ABC):
+    """Abstract class to hold the text we are going to operate on metadata."""
+
+    def __init__(self, start: int, end: int, entity_type: str):
+        self.logger = logging.getLogger("presidio-anonymizer")
+        self.start = start
+        self.end = end
+        self.entity_type = entity_type
+        self.__validate_fields()
+
+    def __gt__(self, other):
+        """Check one entity is greater then other by the text end index."""
+        return self.start > other.start
+
+    def __eq__(self, other):
+        """Check two text metadata entities are equal."""
+        return (self.start == other.start
+                and self.end == other.end
+                and self.entity_type == other.entity_type)
+
+    def __validate_fields(self):
+        validate_parameter_exists(self.start, "result", "start")
+        validate_type(self.start, "start", int)
```

Comment:
I believe that 'validate_parameter' does both 'exists' and 'type'

Reply:
not anymore.

Reply:
Sure? 
validate_parameter
validators.py:39

Reply:
yeah... the validator in line 39 checks is None while the exists checks not ...